lib/redlock/client.rb

```diff
@@ -4,6 +4,9 @@
 module Redlock
   include Scripts
 
+  class RedisTooManyFailuresError < StandardError
```

Comment:
If not limiting errors to `Redis::ConnectionError`, then how about something more abstract, like:

```suggestion
  class LockAcquisitionError < StandardError
```